extensions/amp-analytics/0.1/vendors/ppasanalytics.json

```diff
@@ -0,0 +1,38 @@
+{
+  "vars": {
+    "host": "",
+    "website_id": "",
+    "website_hash": "",
+    "pk_cookie_value": "$REPLACE($CLIENT_ID(_pk_id.${website_id}.${website_hash}),\\.[A-Za-z0-9]*\\.[A-Za-z0-9]*\\.[A-Za-z0-9]*\\.[A-Za-z0-9]*\\.,)",
+    "visitor_id_from_cookie": "$IF($MATCH(${pk_cookie_value},amp-,0),$SUBSTR($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($REPLACE($SUBSTR($TOLOWERCASE(${pk_cookie_value}),4,16),g,10),h,11),i,12),j,13),k,14),l,15),m,16),n,17),o,18),p,19),q,1a),r,1b),s,1c),t,1d),u,1e),v,1f),w,20),x,21),y,22),z,23),A,24),B,25),C,26),D,27),E,28),F,29),G,2a),H,2b),I,2c),J,2d),K,2e),L,2f),M,30),N,31),O,32),P,33),Q,34),R,35),S,36),T,37),U,38),V,39),W,3a),X,3b),Y,3c),Z,3d),_,3e),-,3f),0,16),${pk_cookie_value})",
```

Comment:
What are you trying to do here?

Reply:
I'm trying to convert AMP style ID into hexadecimal string. Our backend tracker is expecting only such ID format.

Reply:
I feel that this will not work. IIRC we have a max depth of URL macro replacements. I think instead we could create a macro that turns a string into hex. 

This would be cleaner and much more readable. Would this be ok?

Reply:
It worked on out test site, so we probably didn't hit this limit yet, but function that would turn AMP ID into hexadecimal string would be a lot easier to use, I agree. Can you tell me when we could add something like that? Can I help with it?

Reply:
@micajuine-ho can you update me on the progress regarding new function?

Reply:
@RomanSek Hi, here is an example PR of how you can create a `$HEX` macro, https://github.com/ampproject/amphtml/pull/33383. Please let me know if you need any help or guidance!

Reply:
Hi, we looked into this and generating hex from arbitrary string will be more complicated. Additionally we plan to use this solution for about 6 months yet. After that we'll update backend to accept AMP ID without need for this transformation, so if you're willing to accept this as temporary solution we would like to proceed with the merge.

Let me know if there are any other blockers before we can merge this.